### PR TITLE
search screen

### PR DESCRIPTION
**Pull request by hungld-1302**
## Related Tickets
- [#Task 54291: Search Screen](https://edu-redmine.sun-asterisk.vn/issues/54291)
## WHAT
- Implement Search Screen
## Evidence (Screenshot or Video)
image

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/sildev/tvshows/screen/SearchActivity.kt [hunk 1]
```diff
@@ -1,13 +1,90 @@
...
+import com.sildev.tvshows.data.repository.SearchRepository
+import com.sildev.tvshows.data.repository.source.remote.SearchRemoteDataSource
 import com.sildev.tvshows.databinding.ActivitySearchBinding
+import com.sildev.tvshows.screen.listtvshow.SearchContract
+import com.sildev.tvshows.screen.listtvshow.SearchPresenter
+import com.sildev.tvshows.screen.listtvshow.adapter.TvShowAdapter
+import com.sildev.tvshows.utils.A_SECOND
+import com.sildev.tvshows.utils.KEY_ID_TV_SHOW
 import com.sildev.tvshows.utils.base.BaseActivity
+import java.util.*
```
Comment (minhnn-1279) [inline app/src/main/java/com/sildev/tvshows/screen/SearchActivity.kt]: bỏ wildcard import nhé a :v

File: app/src/main/res/layout/activity_search.xml [hunk 2]
```diff
@@ -4,6 +4,66 @@
...
+        <ImageView
+            android:layout_width="@dimen/_24dp"
+            android:layout_height="@dimen/_24dp"
+            android:src="@drawable/ic_search"
+            app:tint="@color/color_text_icons" />
+    </LinearLayout>
+
+    <TextView
+        android:id="@+id/text_no_result"
+        android:layout_width="0dp"
```
Comment (minhnn-1279) [inline app/src/main/res/layout/activity_search.xml]: lấy resource dp_0 a nhé

File: app/src/main/java/com/sildev/tvshows/screen/listtvshow/SearchPresenter.kt [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        if (key.trim().isEmpty()) {
+            mView?.onSearchSuccess(mutableListOf())
+            mView?.showTextEmpty()
+            mView?.hideProgressLoading()
+        } else {
+            searchRepository.searchByName(object : OnResultListener<MutableList<TVShow>> {
+                override fun onSuccess(data: MutableList<TVShow>) {
+                    mView?.hideProgressLoading()
+                    mView?.onSearchSuccess(data)
+                    if (data.size <= 0) mView?.showTextEmpty()
```
Comment (minhnn-1279) [inline app/src/main/java/com/sildev/tvshows/screen/listtvshow/SearchPresenter.kt]:
```suggestion
                    if (data.isEmpty()) mView?.showTextEmpty()
```

File: app/src/main/res/layout/activity_search.xml [hunk 4]
```diff
@@ -4,6 +4,66 @@
...
+        android:layout_height="wrap_content"
+        android:gravity="center_vertical"
+        android:orientation="horizontal"
+        android:padding="@dimen/_8dp"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toTopOf="parent">
+
+        <EditText
+            android:id="@+id/edittext_search"
+            android:layout_width="0dp"
```
Comment (ndt0409) [inline app/src/main/res/layout/activity_search.xml]: .

**Pull request merged**